rust/src/graph/builder/mod.rs

```diff
@@ -0,0 +1,59 @@
+use std::path::PathBuf;
+
+use derive_new::new;
+
+use crate::graph::Graph;
+
+#[derive(Debug, Clone, new)]
+pub struct PackageSpec {
+    name: String,
+    directory: PathBuf,
+}
+
+#[derive(Debug, Clone)]
+pub struct GraphBuilder {
+    package: PackageSpec, // TODO(peter) Support multiple packages
+    include_external_packages: bool,
+    exclude_type_checking_imports: bool,
+}
+
+impl GraphBuilder {
+    pub fn new(package: PackageSpec) -> Self {
+        GraphBuilder {
+            package,
+            include_external_packages: false,
+            exclude_type_checking_imports: false,
+        }
+    }
+
+    pub fn include_external_packages(mut self, yes: bool) -> Self {
+        self.include_external_packages = yes;
+        self
+    }
+
+    pub fn exclude_type_checking_imports(mut self, yes: bool) -> Self {
+        self.exclude_type_checking_imports = yes;
+        self
+    }
+
+    pub fn build(&self) -> Graph {
```

Comment:
Added error handling a bit late https://github.com/python-grimp/grimp/pull/263/commits/0c28293a6fd948ec6be152dde280c1a4d6dbad30, would have been better to do this from the start 🙈